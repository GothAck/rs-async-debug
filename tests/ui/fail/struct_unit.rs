use async_debug::AsyncDebug;

#[derive(Debug, AsyncDebug)]
struct Mixed;

#[tokio::main]
async fn main() {
    let mixed = Mixed;

    assert_eq!(
        format!("{:?}", mixed.async_debug().await),
        "Mixed { string: \"test\", integer: 42, rw_lock: [\"string0\", \"string1\"], mutex: [0, 1], mutex_u128: 999 }",
    );
}
