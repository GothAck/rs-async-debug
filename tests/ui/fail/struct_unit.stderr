error: unit structs are not supported
 --> tests/ui/fail/struct_unit.rs:3:17
  |
3 | #[derive(Debug, AsyncDebug)]
  |                 ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `AsyncDebug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `async_debug` found for struct `Mixed` in the current scope
  --> tests/ui/fail/struct_unit.rs:11:31
   |
4  | struct Mixed;
   | ------------- method `async_debug` not found for this
...
11 |         format!("{:?}", mixed.async_debug().await),
   |                               ^^^^^^^^^^^ method not found in `Mixed`
